warehouse/oidc/tasks.py

```diff
@@ -57,3 +58,21 @@ def compute_oidc_metrics(request):
         "warehouse.oidc.total_critical_projects_published_with_oidc_publishers",
         projects_published_with_oidc.where(Project.pypi_mandates_2fa.is_(True)).count(),
     )
+
+    # Metric for total number of files published via OIDC
+    metrics.gauge(
+        "warehouse.oidc.total_files_published_with_oidc_publishers",
+        request.db.query(File.Event)
+        .where(File.Event.additional.op("->>")("publisher_url").is_not(None))
```

Comment:
I can't shake the feeling that there's a better way to perform this query, but I don't have it at hand. I'll continue to muse on it.

Reply:
I guess the question is whether we want the total number of files published via OIDC, or whether we want the total number of files published via oidc in some time frame.

If it's the latter, we can emit metrics for each publish event and let Datadog do the rollup for us, if it's the former... well that's much harder to do without a gauge like this.

Reply:
I think this is what I was looking for:

```suggestion
        .where(File.Event.additional["publisher_url"].is_not(None))
```

The underlying query generates the same syntax, and now we don't have to call the `.op("->>")` semantic, rather we treat the jsonb column like a dict.

If we wanted to use the return value, then we'd pass `.additional["foo"].astext...` but since we're only comparing for non-nulls, and want a count, then we don't need the cast.

Reply:
I completely recognize that we have prior art in the codebase that uses the `op.("->>")` semantic, and we should refactor those to use this, but that's not 100% necessary for this changeset.

Reply:
I tried that, it doesn't work:

```
>>> db.query(File.Event).where(File.Event.additional.op("->>")("publisher_url").is_not(None)).count()
25618

>>> db.query(File.Event).where(File.Event.additional["publisher_url"].is_not(None)).count()
1174355

>>> db.query(File.Event).count()
1174355
```

Reply:
🤔  I guess this is where `.astext` would be the right change, too.